From 1c3a400a7643109ae4cf206031ea3e57a175ad4d Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 13 Mar 2015 14:59:59 +0800
Subject: [PATCH 332/338] atomisp2: Revise dvs coefficients configuration

This patch revises the configuration of dvs coefficients
by setting the coefficients to atomisp_css_params
instead of directly setting to atomisp_css_isp_config.

Change-Id: Iada3ba7b4069bae373c6fe9ec76b258a7e71da9b
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Fixes-Issue: #H1503936301
Reviewed-on: https://icggerrit.ir.intel.com/46222
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: IMIN-legacy (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 3303e00..1bd4688 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -3772,7 +3772,6 @@ void  atomisp_css_set_dvs2_coefs(struct atomisp_sub_device *asd,
 			       struct ia_css_dvs2_coefficients *coefs)
 {
 	asd->params.config.dvs2_coefs = coefs;
-	asd->params.css_update_params_needed = true;
 }
 
 int atomisp_css_set_dis_coefs(struct atomisp_sub_device *asd,
@@ -3828,7 +3827,9 @@ int atomisp_css_set_dis_coefs(struct atomisp_sub_device *asd,
 	    coefs->ver_coefs.even_imag, asd->params.dvs_ver_coef_bytes))
 		return -EFAULT;
 
-	asd->params.config.dvs2_coefs = asd->params.css_param.dvs2_coeff;
+	asd->params.css_param.update_flag.dvs2_coefs =
+		(struct atomisp_dvs2_coefficients *)
+		asd->params.css_param.dvs2_coeff;
 	/* FIXME! */
 /*	asd->params.dis_proj_data_valid = false; */
 	asd->params.css_update_params_needed = true;
-- 
1.9.1

